src/main/java/seedu/toluist/controller/FindController.java

```diff
@@ -0,0 +1,163 @@
+package seedu.toluist.controller;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.logging.Logger;
+
+import seedu.toluist.commons.core.LogsCenter;
+import seedu.toluist.dispatcher.CommandResult;
+import seedu.toluist.model.Task;
+import seedu.toluist.model.TodoList;
+import seedu.toluist.ui.Ui;
+
+/**
+ * Searches the task list for matches in the parameters, and displays the results received
+ */
+public class FindController extends Controller {
+    private static final String COMMAND_TEMPLATE = "^(?<command>(find|filter|list))"
+            + "(\\s+(?<keywords>.+))?\\s*";
+    private static final String SEARCH_BY_TAG = "tag/";
+    private static final String SEARCH_BY_NAME = "name/";
+    private static final String COMMAND_RESULT_TEMPLATE = "Searching for \"%s\" by %s.\n%d results found";
+    private static final String TRUE_COMMAND = "true";
+    private static final String FALSE_COMMAND = "false";
+    private static final String NAME_PARAMETER = "name";
+    private static final String TAG_PARAMETER = "tag";
+    private static final String NAME_AND_TAG_PARAMETER = "name and tag";
```

Comment:
can put all these PARAMETER in front. From now on for constant names I guess we try to keep it standardized with the category name in front. coz @Melvin-Tan was being anal